### PR TITLE
Add docs about CI.

### PR DESCRIPTION
**Pull request by vvbandeira**
Signed-off-by: Vitor Bandeira <vitor.vbandeira@gmail.com>

### CODE REVIEW

File: docs/contrib/CI.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# CI Guide
+
+This document describes the pipelines available to the developers and
+code maintainers. Note that pipelines with the suffix `*-Private` are only
+available to code maintainers and The OpenROAD Project members. To access
```
Comment (maliberty) [inline docs/contrib/CI.md:5]: I would mention why they are private (PDK access) so users see the need for the restriction and don't feel excluded.

File: docs/contrib/CI.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+-   OpenROAD-Coverage-Public
+    -   Description: run dynamic code coverage tool `lconv`.
+    -   Target: master branch.
+    -   Report link here.
+-   OpenROAD-Coverity-Public
+    -   Description: compile and submit builds to Coverity static code analysis
+        tool.
+    -   Target: master branch.
+    -   Report link here.
```
Comment (maliberty) [inline docs/contrib/CI.md:28]: Where is there a report link?

File: docs/contrib/CI.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+-   OpenROAD-Nightly-Public
+    -   Description: `openroad` unit tests, docker builds, ISPD 2018 and 2019
+        benchmarks for DRT and large unit tests of GPL.
+    -   Target: master branch.
+-   OpenROAD-Public
+    -   Description: `openroad` unit tests and docker builds.
+    -   Target: all branches and open PRs.
+-   OpenROAD-Special-Private
+    -   Description: runs ISPD 2018 and 2019 benchmarks for DRT and large unit
```
Comment (maliberty) [inline docs/contrib/CI.md:37]: For developer testing

**Review by maliberty — CHANGES_REQUESTED** [review]

File: docs/contrib/CI.md (hunk 1 above)
Comment (vvbandeira) [reply docs/contrib/CI.md]: Explanation added on [d702c8d](https://github.com/The-OpenROAD-Project/OpenROAD/pull/1740/commits/d702c8dabcbcc16200a3be9db2b96b98ec49c7b9)

File: docs/contrib/CI.md (hunk 2 above)
Comment (vvbandeira) [reply docs/contrib/CI.md]: Fixed on [d702c8d](https://github.com/The-OpenROAD-Project/OpenROAD/pull/1740/commits/d702c8dabcbcc16200a3be9db2b96b98ec49c7b9)

File: docs/contrib/CI.md (hunk 3 above)
Comment (vvbandeira) [reply docs/contrib/CI.md]: Fixed on [d702c8d](https://github.com/The-OpenROAD-Project/OpenROAD/pull/1740/commits/d702c8dabcbcc16200a3be9db2b96b98ec49c7b9)

**Review by maliberty — APPROVED** [review]

**Pull request merged**